### PR TITLE
[storcli] Add 'json' option collecting data in JSON format.

### PR DESCRIPTION
**Pull request by vikas-goel**
Signed-off-by: Vikas Goel <vikas.goel@gmail.com>

---
Please place an 'X' inside each '[]' to confirm you adhere to our [Contributor Guidelines](https://github.com/sosreport/sos/wiki/Contribution-Guidelines)

- [x] Is the commit message split over multiple lines and hard-wrapped at 72 characters?
- [x] Is the subject and message clear and concise?
- [x] Does the subject start with **[plugin_name]** if submitting a plugin patch or a **[section_name]** if part of the core sosreport code?
- [x] Does the commit contain a **Signed-off-by: First Lastname <email@example.com>**?
- [x] Are any related Issues or existing PRs [properly referenced](https://docs.github.com/en/issues/tracking-your-work-with-issues/creating-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) via a Closes (Issue) or Resolved (PR) line?

### CODE REVIEW

**Comment by vikas-goel** [comment]
Default run
```
# sosreport --batch --build -o storcli
...
Running plugins. Please wait ...

  Starting 1/1   storcli         [Running: storcli]

  Finished running plugins


sosreport build tree is located at : /var/tmp/sosreport-xxxx-2021-08-02-gewrokw
# ls /var/tmp/sosreport-xxxx-2021-08-02-gewrokw/sos_commands/storcli/
storcli64_.call.bbu_show_all  storcli64_.call.eall.sall_show_all  storcli64_.call_show_AliLog  storcli64_.call_show_termlog
storcli64_.call.cv_show_all   storcli64_.call.eall_show_all       storcli64_.call_show_all     storcli64_.call.vall_show_all
storcli64_.call.dall_show     storcli64_.call.sall_show_all       storcli64_.call_show_events  storcli64_show_ctrlcount
# cat /var/tmp/sosreport-xxxx-2021-08-02-gewrokw/sos_commands/storcli/storcli64_show_ctrlcount
CLI Version = 007.0813.0000.0000 Dec 14, 2018
Operating system = Linux 3.10.0-1062.26.1.el7.x86_64
Status Code = 0
Status = Success
Description = None

Controller Count = 1


#
```

With JSON option
```
# sosreport --batch --build -o storcli -k storcli.json=true
...
Running plugins. Please wait ...

  Starting 1/1   storcli         [Running: storcli]

  Finished running plugins


sosreport build tree is located at : /var/tmp/sosreport-xxxx-2021-08-02-kcqelzf


# ls /var/tmp/sosreport-xxxx-2021-08-02-kcqelzf/sos_commands/storcli/
storcli64_.call.bbu_show_all_J  storcli64_.call.eall.sall_show_all_J  storcli64_.call_show_AliLog_J  storcli64_.call_show_termlog_J
storcli64_.call.cv_show_all_J   storcli64_.call.eall_show_all_J       storcli64_.call_show_all_J     storcli64_.call.vall_show_all_J
storcli64_.call.dall_show_J     storcli64_.call.sall_show_all_J       storcli64_.call_show_events_J  storcli64_show_ctrlcount_J
# cat /var/tmp/sosreport-xxxx-2021-08-02-kcqelzf/sos_commands/storcli/storcli64_show_ctrlcount_J
{
"Controllers":[
{
        "Command Status" : {
                "CLI Version" : "007.0813.0000.0000 Dec 14, 2018",
                "Operating system" : "Linux 3.10.0-1062.26.1.el7.x86_64",
                "Status Code" : 0,
                "Status" : "Success",
                "Description" : "None"
        },
        "Response Data" : {
                "Controller Count" : 1
        }
}
]
}


#
```

**Comment by vikas-goel** [comment]
Hi @TurboTurtle , @pmoravec ,
Please review.

File: sos/report/plugins/storcli.py [hunk 1]
```diff
@@ -36,9 +40,14 @@ def setup(self):
 
         logpath = self.get_cmd_output_path()
 
+        json = ''
+        if self.get_option('json'):
+            json = ' J'
```
Comment (pmoravec) [inline sos/report/plugins/storcli.py:45]: Maybe just subjective stylistic option: can't it be shortened to

    json = ' J' if self.get_option('json') else ''

?

**Review by pmoravec — APPROVED** [review]
ACK also the current code.

File: sos/report/plugins/storcli.py (hunk 1 above)
Comment (TurboTurtle) [reply sos/report/plugins/storcli.py:45]: I actually think the one-liner is more readable, personally.

+1 for shortening, then ack from me.

Comment (vikas-goel) [reply sos/report/plugins/storcli.py]: @TurboTurtle ,
Incorporated the change. PTAL.

**Review by pmoravec — APPROVED** [review]

**Review by TurboTurtle — APPROVED** [review]

**Pull request merged**